Orange/widgets/unsupervised/tests/test_owkmeans.py

```diff
@@ -0,0 +1,43 @@
+from PyQt4 import QtGui
+
+from Orange.widgets.tests.base import WidgetTest
+from Orange.widgets.unsupervised.owkmeans import OWKMeans
+
+from Orange.data import Table
+class TestOWKMeans(WidgetTest):
+
+    def setUp(self):
+        self.widget = self.create_widget(OWKMeans)
+        self.combo_box = self.widget.findChildren(QtGui.QComboBox)
+        self.spinner = self.widget.findChildren(QtGui.QSpinBox)
+        self.radio_button = self.widget.findChildren(QtGui.QRadioButton)
+        self.iris = Table("iris")
+
+    def test_boxes_visible(self):
+        """ Check if all boxes visible """
+        self.assertEqual(self.combo_box[0].isHidden(), False)
+        self.assertEqual(self.combo_box[1].isHidden(), False)
+        self.assertEqual(self.combo_box[2].isHidden(), False)
+        self.assertEqual(self.spinner[0].isHidden(), False)
+        self.assertEqual(self.spinner[1].isHidden(), False)
+        self.assertEqual(self.spinner[2].isHidden(), False)
+
+
+    def test_optimization_report_display(self):
+        """ Check visibility of optimization report after we select the number of clusters  """
+        for button_pos, value in ((0, True),
```

Comment:
I would rewrite this without a for, so it will look like a "scenario". Something like:

1. Select fix number of clusters
2. Check that optimization part is hidden
3. Select optimized number of clusters
4. Check that optimization part is shown
5. Click apply
6. Check that some lines were added to the optimization pane